provider/spark.go

```diff
@@ -175,6 +177,28 @@ func (q defaultSparkOfflineQueries) trainingSetCreate(def TrainingSetDef, featur
 		featureJoinQuery := fmt.Sprintf("LEFT OUTER JOIN (%s) t%d ON (t%d_entity = entity AND t%d_ts <= label_ts)", featureWindowQuery, i+1, i+1, i+1)
 		joinQueries = append(joinQueries, featureJoinQuery)
 	}
+	for i, lagFeature := range def.LagFeatures {
```

Comment:
Is this what pandas is using?

Reply:
yeah, we are using the Spark SQL Queries for it. And the training set will be modified to include lag features.